internal/service/searchindex/resource_search_index_test.go

```diff
@@ -316,32 +225,35 @@ func TestAccSearchIndexRS_withVector(t *testing.T) {
 		Steps: []resource.TestStep{
 			{
 				Config: configVector(clusterInfo.ProjectIDStr, indexName, databaseName, clusterInfo.ClusterNameStr, clusterInfo.ClusterTerraformStr),
-				Check: resource.ComposeTestCheckFunc(
-					checkExists(resourceName),
-					resource.TestCheckResourceAttr(resourceName, "name", indexName),
-					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
-					resource.TestCheckResourceAttr(resourceName, "cluster_name", clusterInfo.ClusterName),
-					resource.TestCheckResourceAttr(resourceName, "database", databaseName),
-					resource.TestCheckResourceAttr(resourceName, "collection_name", collectionName),
-					resource.TestCheckResourceAttr(resourceName, "type", "vectorSearch"),
-					resource.TestCheckResourceAttrSet(resourceName, "fields"),
-					resource.TestCheckResourceAttrWith(resourceName, "fields", acc.JSONEquals(fieldsJSON)),
-
-					resource.TestCheckResourceAttr(datasourceName, "type", "vectorSearch"),
-					resource.TestCheckResourceAttr(datasourceName, "name", indexName),
-					resource.TestCheckResourceAttrSet(datasourceName, "project_id"),
-					resource.TestCheckResourceAttr(datasourceName, "cluster_name", clusterInfo.ClusterName),
-					resource.TestCheckResourceAttr(datasourceName, "database", databaseName),
-					resource.TestCheckResourceAttr(datasourceName, "collection_name", collectionName),
-					resource.TestCheckResourceAttr(datasourceName, "name", indexName),
-					resource.TestCheckResourceAttrSet(datasourceName, "index_id"),
-					resource.TestCheckResourceAttrSet(datasourceName, "fields"),
-					resource.TestCheckResourceAttrWith(datasourceName, "fields", acc.JSONEquals(fieldsJSON)),
-				),
+				Check:  resource.ComposeTestCheckFunc(checks...),
 			},
 		},
 	})
 }
+func commonChecks(indexName, indexType, mappingsDynamic, databaseName string, clusterInfo acc.ClusterInfo) []resource.TestCheckFunc {
```

Comment:
NOTE: adds `AddAttrChecks` and `AddAttrSetChecks` to `acc` package

These functions are meant to simplify the reusing of checks.
This function is used by almost all of the tests to simplify the checking.
It defines:

1. `attributes`used by `AddAttrChecks` the common fields to be checked on datasource and resource for all tests (checking both field and value)
2. `AddAttrSetChecks` only checks that the field is defined
3. any differences between `resourceName` and `datasourceName`, e.g., `index_id` is also checked on the `datasourceName`

In this case, I add two more helpers to simplify adding checks to both datasource and resource: `addAttrChecks` and `addAttrSetChecks`